torchvision/prototype/transforms/functional/_color.py

```diff
@@ -79,9 +79,14 @@ def adjust_contrast_image_tensor(image: torch.Tensor, contrast_factor: float) ->
     c = image.shape[-3]
     if c not in [1, 3]:
         raise TypeError(f"Input image tensor permitted channel values are {[1, 3]}, but found {c}")
-    dtype = image.dtype if torch.is_floating_point(image) else torch.float32
-    grayscale_image = _rgb_to_gray(image) if c == 3 else image
-    mean = torch.mean(grayscale_image.to(dtype), dim=(-3, -2, -1), keepdim=True)
+    fp = image.is_floating_point()
+    if c == 3:
+        grayscale_image = _rgb_to_gray(image, cast=False)
+        if not fp:
+            grayscale_image = grayscale_image.floor_()
```

Comment:
@pmeier In an early iteration of the PR, I have missed this `floor_()` call which is necessary to reproduce identical results with stable. Unfortunately the tests passed without catching the issue. Might be worth checking that our reference tests check both on floats and ints.

Reply:
This is properly tested with 

https://github.com/pytorch/vision/blob/f32600b66d9c03682ccd975e442e67c609fd7f0a/test/prototype_transforms_kernel_infos.py#L1845-L1850

but it seems our tolerances are to high to pick up on this

https://github.com/pytorch/vision/blob/f32600b66d9c03682ccd975e442e67c609fd7f0a/test/prototype_transforms_kernel_infos.py#L1866

https://github.com/pytorch/vision/blob/f32600b66d9c03682ccd975e442e67c609fd7f0a/test/prototype_transforms_kernel_infos.py#L64-L67

However, we can't remove the tolerances completely here, since the output differs even with the current implementation.

Reply:
Hmm... 99% of the performance optimizations were tested for training and we know they are correct. So there might be rounding error issues. I think we should adjust the tolerances to the degree that will catch BC issues but not go ballistic for the 8th decimal. That might be a hard exercise though.